File: prova-test-softwre.py
#Este bloco importa todas as bibliotecas usadas posteriormente no código
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
from webdriver_manager.chrome import ChromeDriverManager
import time

#inicializa o chrome e maximiza janela
driver = webdriver.Chrome(service=Service(ChromeDriverManager().install()))
driver.maximize_window()

try: #captura erros na execução 
    url = "https://www.hankeds.com.br/prova/login.html" #URL
    driver.get(url) #URL

    time.sleep(2) #Espera

    def digitar_lento(elemento, texto, delay=0.25): #delay na digitação
        for letra in texto:
            elemento.send_keys(letra)
            time.sleep(delay)

    usuario = WebDriverWait(driver, 10).until(EC.presence_of_element_located((By.ID, "username"))) #acha username por ID
    senha = driver.find_element(By.ID, "password")#acha password por ID
    botao = driver.find_element(By.XPATH, "//button[contains(text(), 'Entrar')]") # acha o botão por xpath

    digitar_lento(usuario, "admin") #admin respeitando a espera
    time.sleep(1) #espera
    digitar_lento(senha, "admin123456") #digita senha respeitando a espera
    time.sleep(1)#espera

    botao.click() #clica no botão especificado
    time.sleep(4)#espera

    if "destino.html" in driver.current_url: #confirma o destino e se deu certo
        print(" Teste passou: redirecionado corretamente.")
    else:
        print(" Teste falhou: redirecionamento não ocorreu.")

    time.sleep(5)#espera

except Exception as e: #se teve erro no teste print do erro
    print(" Erro durante o teste:", str(e))

finally:
    driver.quit()

File: prova-test-softwre.js
const { Builder, By } = require('selenium-webdriver'); //importa os modulos necessesarios

//Função testehanked
(async function testeHanked() {
const url = 'https://www.hankeds.com.br/prova/login2.html'; //define URL
let driver = await new Builder().forBrowser('chrome').build(); //Inicia chromedriver

try {

await driver.get(url); //Acessa pagina
await driver.sleep(2000); //espera

//localiza os campos usuario e senha por id e o botão por xpath
const username = await driver.findElement(By.id('username'));
const password = await driver.findElement(By.id('password'));
const botao = await driver.findElement(By.xpath("//button[contains(text(),'Entrar')]"));

// digita admin esperando entre as letras
for (const letra of 'admin') {
await username.sendKeys(letra);
await driver.sleep(250);
}

await driver.sleep(1000); // espera

// digita senha esperando entre as letras
for (const letra of 'admin123456') {
await password.sendKeys(letra);
await driver.sleep(250);
}

// espera e clica no botão
await driver.sleep(1000); 
await botao.click();

await driver.sleep(4000); //espera

const urlAtual = await driver.getCurrentUrl(); // pega a URL da pagina

//faz a verificação do teste
if (urlAtual.includes('destino.html')) {
console.log(' Teste passou: redirecionado corretamente.');
} else {
console.log(' Teste falhou: não houve redirecionamento.');
}

await driver.sleep(5000); //espera

//se houver erro avisa no console
} catch (err) {
console.error(' Erro durante o teste:', err);
} finally {
await driver.quit();
}
})();